RUFAS/biophysical/animal/animal.py

```
@@ -2109,6 +2165,316 @@ def _feedlot_life_stage_update(self, time: RufasTime) -> tuple[AnimalStatus, New
+        self.events.add_event(self.days_born, time.simulation_day, animal_constants.CALF_WEANED)
+
+        if self.dam is not None:
+            self.dam.calf_at_side = None
+
+        destination = AnimalConfig.beef_post_weaning_destination
+        if destination == "sell":
+            self.sold_at_day = time.simulation_day
+            return AnimalStatus.SOLD, None
+        if destination == "replacement_heifer":
+            if self.sex != Sex.FEMALE:
+                raise ValueError(f"Calf with sex {self.sex} cannot be transitioned to a replacement heifer.")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)